dfget/core/api/supernode_api.go

```diff
@@ -90,7 +93,9 @@ func (api *supernodeAPI) PullPieceTask(node string, req *types.PullPieceTaskRequ
 		api.Scheme, node, peerPullPieceTaskPath, util.ParseQuery(req))
 
 	resp = new(types.PullPieceTaskResponse)
-	e = api.get(url, resp)
+	if e = api.get(url, resp); e != nil {
```

Comment:
Why to change this? I do not see any difference after the change actually.

Reply:
https://github.com/dragonflyoss/Dragonfly/blob/c5d08731c856c3f71d09eb0a0dc7567754f7a0ed/dfget/core/downloader/p2p_downloader/p2p_downloader.go#L240

Reply:
if the supernode is off,the resp={} in the previous code.and res.code will make the dfget happen a `a invalid memory address or nil pointer dereference` err

Reply:
I return a nil if the supernode if off so the the program will  go on

Reply:
It is so tricky. I would like to see you change the function definition. Please do not define the variables in the function declaration. Just define them in the implementation block.

Reply:
> It is so tricky. I would like to see you change the function definition. Please do not define the variables in the function declaration. Just define them in the implementation block.
In fact I can't understand your  comment very clearly,Can in more details?.But I think the coder want to return a nil when a err happen such as https://github.com/dragonflyoss/Dragonfly/blob/c5d08731c856c3f71d09eb0a0dc7567754f7a0ed/dfget/core/api/supernode_api.go#L73.
and he just ignore this place in the previous implement .

Reply:
> It is so tricky. I would like to see you change the function definition. Please do not define the variables in the function declaration. Just define them in the implementation block.
In fact I can't understand your  comment very clearly,Can in more details?.But I think the coder want to return a nil when a err happen such as https://github.com/dragonflyoss/Dragonfly/blob/c5d08731c856c3f71d09eb0a0dc7567754f7a0ed/dfget/core/api/supernode_api.go#L73.
and he just ignore this place in the previous implement .